config/config.go

```diff
@@ -388,23 +380,31 @@ func refreshRemoteList(
 				err = readRemoteList(raw, gl.RemoteConfig, remotes, maxNumSrcTCPPorts,
 					minBatchInterval, logger)
 				if err != nil {
-					logger.Error("error parsing downloaded YAML configuration file",
+					logger.Error("error parsing downloaded configuration file",
 						zap.Error(err))
-					goto cont
+					goto contError
 				}
+				gl.Remotes = remotes
 				logger.Info("Will poll Orchestrator again later",
 					zap.String("retry_time",
 						gl.RemoteConfig.PollOrchestratorInterval.Success.String()))
 				return nil
 
 			case http.StatusNotFound:
 				retryTime = gl.RemoteConfig.PollOrchestratorInterval.Success
-				goto cont
+				goto stayIdle
```

Comment:
If there is an issue obtaining the configuration, the execution continues at line 398 and if a config has been previously fetched successfully code enters if statement of line 401. A default case, I believe, is not needed since for non-errored fetch, one of the two responses is expected.